### DISCUSSION TITLE
Potential roadmap items?

### DISCUSSION THREAD

**Discussion by akhanf** [khanlab/snakebids#70] [General]
Should start collating these here, then can add to kanban board

**Comment by pvandyken** [comment]
Move default CLI args internally. Allow for definition of args in run.py

**Comment by akhanf** [comment]
full conformance with snakemake "best practice" workflows

**Comment by akhanf** [comment]
better tutorials?
data carpentry snakebids?

**Reply by kaitj** [reply]
Here is the example of the snakemake data carpentry tutorial: https://github.com/carpentries-incubator/snakemake-novice-bioinformatics

**Comment by akhanf** [comment]
end-to-end tests

**Comment by akhanf** [comment]
deployment instructions/template

**Reply by akhanf** [reply]
snakemake recommended conda+singularity?
alternatives to conda (Peter's venv fix)
hippunfold-style build (all python deps in the project)

**Comment by akhanf** [comment]
adding to the project-templates
- more config options?
- custom templates?

**Comment by akhanf** [comment]
chatted about this briefly at a previous meeting, but would be good to add more clarity on the zip lists (and related filter functions) vs regular lists, when you need them etc. 

Or maybe an overhaul of how that was done, using a better data structure than multiple lists zipped together -- e.g. could make a dataframe (a row for each bids instance -- I wonder if pybids just does this anyways?) - I haven't thought about how that would work with the bids() function though..
